### ISSUE TITLE
GBG Logs - Sector building costs > output to text

### ISSUE THREAD

**Issue by Valutin** [mainIine/foe-helfer-extension#2384]
Hello, this is my first request and I would like to apologize if it comes in a form that is not intended here, actually my first post on github..

For my guild, I keep track of our GBG building expenses and I use the info to forecast and warn my guildmates about what type of goods we will have in need of extra donations on top of what their buildings generate., For example, if we have 100k cloth in stock and after 2 days of GBG swaps, we are consuming 15k per day with an income of 5k, then, I'll end our GBG season well below our stock warning of 75k. I'll then tell my team to start generating or trading more cloth to fill stock up. It's a sligthly more proactive approch than, we reached our safety level, we stop building anything. 

So I browse the log, check and record when and what type of building have been built, then copy the info into a spreadsheet. The spreadsheet contains guild data, (gather and combine by foe-helper, which I am very very thankful for).
My question would be if it is possible for the extension to generate the log into a text string or a table that I can manipulate? Like the treasurry that I can copy/paste straight from the foe-helper extension. 

For my spreadsheet, I record:
-sector name and time (in 24h format). 
-I also manually record cost of watchtower/siege camp by manually visiting each conquered sector.
Those are the two items I would like to know if it's possible to have access from foe-helper via a copy/paste action. 

Thanks in advance for the time you will take to read this, if you need info about my own spreadsheet. I can share that, but it's not really in a publishing state. As usual, only the spreadsheet owner knows what he is doing.. :)

**Comment by outoftheline** [comment]
Do you check the log regularly? Unfortunately only the latest 200 actions are tracked in it, so really active guilds might fill that up quite quickly.

**Comment by Valutin** [comment]
Sorry for the late reply, Yes, I do check the log regularly, like at least once every 12h. The game seems to keep everything up to 12h. 
This is a link to my spreadsheet, GBG just started on Volcano map, so I have to record map data (sheet "Volcano Map Data", I have a separate one for Waterfall), then, add each and every single SC built in "Record" (time stamp, sector and the resources filled automatically from the map data sheet). 
I will fill once both sheets "Guild Treasury" and "Guildmates Buildings" with FoeHelper's Guild function help, I go and visit all my guildmates city and then visit the treasury page. 

With all that, I build GBG cost sheet (2 actually) which is just a pivot table of our cost per day, comparing it with daily production and remaining stock, I will post on the guild thread a message if some of our goods are predicted to fall under a set alarm level (like 100k goods for goods <FE).

https://docs.google.com/spreadsheets/d/1AQ2o_74RpfCP6Anje5S8tBDIEkMXi-c18LiGYg4ssEU/edit?usp=sharing

**Comment by 3lackghost** [comment]
I created a script for my PC that, after importing a CSV file from the guild treasury, shows me which goods have dropped below a certain limit I set and sends the results to our Discord server

**Comment by panda667** [comment]
Oooh nice! What about getting the info directly from FoE helper? Then you don't need to download csvs constantly.

When it works, upload it to github and maybe it'll get implemented in a future version of FoE helper!

**Comment by 3lackghost** [comment]
> Oooh nice! What about getting the info directly from FoE helper? Then you don't need to download csvs constantly.
> 
> When it works, upload it to github and maybe it'll get implemented in a future version of FoE helper!

Development is underway, now implemented in JavaScript. I have the 'FoE Treasury Monitor Version 3.5 exe' executable ready on my machine, along with a demo video of the tool in action."

**Comment by panda667** [comment]
That is amazing. I wish I could do stuff like that.

**Comment by 3lackghost** [comment]
> That is amazing. I wish I could do stuff like that.

Here is a video of the program working.

https://youtu.be/thWP_XXwTUw?si=W4Mz2YlSeX4CjyfA

**Comment by 3lackghost** [comment]
https://github.com/3lackghost/foe-treasury-monitor/releases/tag/v3.8

**Comment by outoftheline** [comment]
@Arklur was working on the logs for GBG, but he’s not finished yet afaik

**Comment by Arklur** [comment]
That's correct, didn't really have the time in the recent weeks to make the "finishing touches".
